src/librbd/io/ImageRequestWQ.cc

```diff
@@ -557,6 +588,14 @@ void *ImageRequestWQ<I>::_void_dequeue() {
     ThreadPool::PointerWQ<ImageRequest<I> >::_void_dequeue());
   assert(peek_item == item);
 
+  if (item->needs_throttle() && iops_throttle->get(1, new C_Throttle(this, item))) {
+    // io was queued into blockers list and wait for tokens.
+    this->get_pool_lock().Unlock();
```

Comment:
Don't signal the thread -- you are pausing here by returning ```nullptr``` and ```C_Throttle``` should wake it back up (busy spin loop)

Reply:
I don't want to block the remain IOs, if we only set a write iops limit, and there are still read IO in the queue, I want to signal the thread to continue.

Reply:
... but right now this causes a busy-loop since it will constantly be waking up the thread to look at the first item again, see that it's throttled, and go back to sleep -- to wake itself up again.

Reply:
This is not a busy-loop, the next loop will look at the second item, because the current item is already dequeued by ```ThreadPool::PointerWQ<ImageRequest<I> >::_void_dequeue()```.

Reply:
... but not required for this PR w/ a global throttle since once requests are throttled, IO should stop